pennylane/optimize/rotosolve.py

```diff
@@ -132,340 +212,410 @@ class RotosolveOptimizer:
+    - ``layer_par`` feeds into a layer of rotations with a single parameter, and
+    - ``crot_param`` feeds three parameters, multiplied with ``crot_weights``, into
+      three controlled Pauli rotations.
 
-    We also initialize a set of parameters for all these operations, and summarize
-    the numbers of frequencies in ``num_freqs``.
+    We also initialize a set of parameters for all these operations, and start with
+    uniform weights, i.e. all ``rot_weights`` and ``crot_weights`` are set to one.
+    This means that all frequencies with which the parameters in ``rot_param`` and
+    ``crot_param`` enter the QNode are integer-valued.
+    The number of frequencies per parameter are summarized in ``nums_frequency``.
 
     .. code-block :: python
 
-        init_param = [
+        init_param = (
             np.array([0.3, 0.2, 0.67], requires_grad=True),
             np.array(1.1, requires_grad=True),
             np.array([-0.2, 0.1, -2.5], requires_grad=True),
-        ]
+        )
+        rot_weights = np.ones(3)
+        crot_weights = np.ones(3)
 
-        num_freqs = [[1, 1, 1], 3, [2, 2, 2]]
+        nums_frequency = {
+            "rot_param": {(0,): 1, (1,): 1, (2,): 1},
+            "layer_par": {(): 3},
+            "crot_param": {(0,): 2, (1,): 2, (2,): 2},
+        }
 
     The keyword argument ``requires_grad`` can be used to determine whether the respective
     parameter should be optimized or not, following the behaviour of gradient computations and
-    gradient-based optimizers when using Autograd.
+    gradient-based optimizers when using Autograd or Torch.
 
-    In addition, the optimization technique for the Rotosolve substeps can be chosen via the
-    ``optimizer`` and ``optimizer_kwargs`` keyword arguments.
-    As an extra feature, the minimized cost of the intermediate univariate reconstructions can
+    Now we carry out the optimization.
+    The minimized cost of the intermediate univariate reconstructions can
     be read out via ``full_output``, including the cost *after* the full Rotosolve step:
 
-    .. code-block :: python
-
-        param = init_param.copy()
-        cost_rotosolve = []
-        for step in range(num_steps):
-            param, cost, sub_cost = opt.step_and_cost(
-                cost_function,
-                *param,
-                num_freqs=num_freqs,
-                full_output=True,
-            )
-            print(f"Cost before step: {cost}")
-            print(f"Minimization substeps: {np.round(sub_cost, 6)}")
-            cost_rotosolve.extend(sub_cost)
-
-    The optimized values for ``x`` are now stored in ``param`` and (sub)steps-vs-cost can be
-    assessed by plotting ``cost_rotosolve``.
+    >>> param = init_param
+    >>> cost_rotosolve = []
+    >>> for step in range(num_steps):
+    ...     param, cost, sub_cost = opt.step_and_cost(
+    ...         cost_function,
+    ...         *param,
+    ...         nums_frequency=nums_frequency,
```

Comment:
is this still preserved for backwards compatibility?

Reply:
The `nums_frequency` kwarg is an up-to-date keyword argument for `reconstruct`, so it also is one for `Rotosolve` :)